### ISSUE TITLE
Help output should go to STDOUT

### ISSUE THREAD

**Issue by eikenb** [hashicorp/vault#17004]
**Describe the bug**

Vault help should be written to STDOUT, not STDERR.

**To Reproduce**
Don't work (and should).
$ vault -h | less
$ vault read -h | most
$ vault write -h | view -

Workaround.
$ vault -h 2>&1 | less

**Expected behavior**
To be able to view help in pager without having to redirect STDERR.

**Environment:**
vault v1.11.2, Linux

Vault server configuration file(s):
N/A

**Additional context**
Standard CLI behavior is to output help on STDOUT when requested as it is not an error.

**Comment by cipherboy** [comment]
Interestingly, while [GNU standards do say this](https://www.gnu.org/prep/standards/html_node/_002d_002dhelp.html) (see: `on standard output`), non-GNU programs like `openssl` and others consistently write help to stderr. As such, this is probably not a bug, but a feature request.

In the past, it seems like there might've been an intent to allow users to customize this behavior via `RunCustom`, but I don't see any existing options to do so. From the core shared CLI code, it does support setting it to your choice of writer:

https://github.com/mitchellh/cli/blob/25f4c7ceb617dfe9729b3ff32be02f7544ff7fb4/cli.go#L119-L123

but Vault explicitly overrides this: 

https://github.com/hashicorp/vault/blob/d46c655e6ed34da5030c46a78ee2862e7989e6be/command/main.go#L231

But its not clear as to why. It was at a time before `ErrorWriter` existed in cli...


Regardless, I'm inclined to believe this is a breaking change and probably needs more discussion; existing scripts may break based on this behavior.

**Comment by eikenb** [comment]
The reason why I filed this as a bug is that the behavior is bad, particularly with long help output like many hashicorp products have. It is bad UX to have to redirect STDERR every time you want to pipe the help to a pager so you can read it using software intended for that purpose instead of having to rely on scrollback buffers.

To put it more in terms of a feature... help to STDOUT lets you pipe it to a pager for easy reading providing a better UX. Help going to STDERR has no advantages and has a worse UX. Help would be better going to STDOUT.

**Comment by cipherboy** [comment]
Oh, I definitely agree with you, but at this point, Vault has had this behavior for N>5 years. This is something that is probably baked into the assumptions of scripts and other automation systems at this point.

I worry that the only way this gets accepted is with an environment variable change (like what occurred with detailed LIST behavior #15417 or is also occurring with 404 not found/#14944's solution). _Maybe_ it gets in with interactive TTY detection, but...

**Comment by eikenb** [comment]
I agree that backwards compatibility is **very** important, but  changing explicitly called help output should not impact automation as automation doesn't call help. IMO the UX benefit is enough to overcome the *remote* chance that help output automation is part of a critical path. 

Not my call though and I appreciate your considering it.

Thanks.

**Comment by cipherboy** [comment]
Ah I see, the `HelpWriter` is strictly used for a case when `-help` exists on flags given to the program. `ErrorWriter` is used when Help is printed as part of Usage (i.e., bad flags, the case I was mostly worried about -- if we have bad flags and we print the help to stdout suddenly rather than stderr, that'd be bad).

In that sense, I think I'm fine fixing this. I'll propose a PR.
